pkg/virt-controller/watch/drain/evacuation/evacuation.go

```diff
@@ -369,6 +369,13 @@ func (c *EvacuationController) sync(node *k8sv1.Node, vmisOnNode []*virtv1.Virtu
 		Effect: k8sv1.TaintEffectNoSchedule,
 	}
 
+	activeMigrationsFromThisSourceNode := 0
+	for _, vmi := range vmisOnNode {
+		if migrationutils.IsMigrating(vmi) {
+			activeMigrationsFromThisSourceNode++
+		}
+	}
+
```

Comment:
Can you move this code so it would occur after this section?

```go
	// Don't create hundreds of pending migration objects.
	// This is just best-effort and is *not* intended to not overload the cluster.
	// It is possible that more migrations than the limit are created because of evacuations on other nodes.
	// The migration controller needs to limit itself to a reasonable number of running migrations
	maxParallelMigrations := int(*c.clusterConfig.GetMigrationConfiguration().ParallelMigrationsPerCluster)
	if len(activeMigrations) >= maxParallelMigrations {
		// We have to re-enqueue if some work is left, since migrations from other controllers or workers` don't wake us up again
		if len(migrationCandidates) > 0 || len(nonMigrateable) > 0 {
			c.Queue.AddAfter(node.Name, 5*time.Second)
		}
		return nil
	}
```

If you would return in the above code, this calculation isn't needed

Reply:
Nice catch! will remove